agent/connect/ca/provider_vault.go

```diff
@@ -437,7 +438,7 @@ func (v *VaultProvider) Sign(csr *x509.CertificateRequest) (string, error) {
 		return "", fmt.Errorf("issuing_ca was not a string")
 	}
 
-	return fmt.Sprintf("%s\n%s", cert, ca), nil
+	return fmt.Sprintf("%s\n%s\n", AddSingleNewline(cert), AddSingleNewline(ca)), nil
```

Comment:
Don't we have an extra trailing newline here now, because the format also has an extra `\n` ?

Reply:
nice catch!!